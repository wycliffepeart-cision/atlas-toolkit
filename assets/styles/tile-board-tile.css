@layer atlas {
	atlas-tile-board-tile {
		--atlas-title-board-tile-border-radius: 0.5em;
		--atlas-title-board-padding: 1em;

		display: flex;
		flex-flow: nowrap;

		background-color: var(--atlas-color-white);
		padding: var(--atlas-title-board-padding);
		box-shadow: var(--atlas-tile-board-tile-shadow);
		border-radius: var(--atlas-title-board-tile-border-radius);
	}

	atlas-tile-board-tile::part(controls) {
		display: block;
	}

	atlas-tile-board-tile::part(body) {
		display: flex;
		flex-wrap: wrap;
		flex-grow: max-content;
		width: 100%;
	}
}
