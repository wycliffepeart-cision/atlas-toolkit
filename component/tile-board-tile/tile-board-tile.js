import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var TileBoardTileTemplate = "<template>\n  <slot part=\"body\"></slot>\n  <slot name=\"controls\" part=\"controls\"></slot>\n</template>\n";

let TileBoardTile = class TileBoardTile extends HTMLElement {
};
TileBoardTile = __decorate([
    Template(TileBoardTileTemplate)
], TileBoardTile);
if (!customElements.get('atlas-tile-board-tile'))
    customElements.define('atlas-tile-board-tile', TileBoardTile);

export { TileBoardTile };
