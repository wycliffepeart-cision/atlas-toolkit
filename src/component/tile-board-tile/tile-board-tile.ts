import { Template } from '../../decorators/template';
import TileBoardTileTemplate from './tile-board-tile.html';

@Template(TileBoardTileTemplate)
export class TileBoardTile extends HTMLElement {}

if (!customElements.get('atlas-tile-board-tile')) customElements.define('atlas-tile-board-tile', TileBoardTile);
