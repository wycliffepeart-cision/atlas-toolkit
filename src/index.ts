export * from './component/accordion/accordion';
export * from './component/alert/alert';
export * from './component/application-frame/application-frame';
export * from './component/button/button';
export * from './component/icon/icon';
export * from './component/button-set/button-set';
export * from './component/chip/chip';
export * from './component/chip-group/chip-group';
export * from './component/dialog/dialog';
export * from './component/feed/feed';
export * from './component/input-checkbox/input-checkbox';
export * from './component/input-chip-group/input-chip-group';
export * from './component/input-dropdown/input-dropdown';
export * from './component/input-search/input-search';
export * from './component/input-text/input-text';
export * from './component/input-textarea/input-textarea';
export * from './component/logo/logo';
export * from './component/main-nav/main-nav';
export * from './component/main-nav-item/main-nav-item';
export * from './component/number-display/number-display';
export * from './component/profile-control/profile-control';
export * from './component/snackbar/snackbar';
export * from './component/stepper/stepper';
export * from './component/stepper-step/stepper-step';
export * from './component/tile-board/tile-board';
export * from './component/tile-board-tile/tile-board-tile';
export * from './component/workflow/workflow';
export * from './component/workflow-step/workflow-step';
