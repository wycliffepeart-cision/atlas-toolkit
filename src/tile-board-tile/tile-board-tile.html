<template>
  <slot part="body"></slot>
  <slot name="controls" part="controls"></slot>
</template>
