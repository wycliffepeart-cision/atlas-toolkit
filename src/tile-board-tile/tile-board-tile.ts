import { Template } from '../decorators/template';
import TileBoardTileTemplate from './tile-board-tile.html';

@Template(TileBoardTileTemplate)
export class TileBoardTile extends HTMLElement {}
