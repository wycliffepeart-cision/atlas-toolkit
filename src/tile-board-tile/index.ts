import { TileBoardTile } from './tile-board-tile';

if (!customElements.get('atlas-tile-board-tile')) customElements.define('atlas-tile-board-tile', TileBoardTile);
