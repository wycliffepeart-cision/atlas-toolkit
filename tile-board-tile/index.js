import { TileBoardTile } from './tile-board-tile.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-tile-board-tile'))
    customElements.define('atlas-tile-board-tile', TileBoardTile);
