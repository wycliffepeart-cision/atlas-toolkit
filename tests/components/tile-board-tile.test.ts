import '../../src/tile-board-tile';

const TILE_BOARD_TILE_TAG = 'atlas-tile-board-tile';
const getShadowDomContent = (element) => element.shadowRoot.innerHTML;

describe('Tile Board', () => {
  it('atlas-tile-board-tile exists', async () => {
    await customElements.whenDefined(TILE_BOARD_TILE_TAG);
    const dashboardElement = document.createElement(TILE_BOARD_TILE_TAG);
    document.body.appendChild(dashboardElement);

    (expect(dashboardElement) as any).toBeInTheDocument();
  });

  it('atlas-tile-board-tile selectable', async () => {
    await customElements.whenDefined(TILE_BOARD_TILE_TAG);
    const dashboardElement = document.createElement(TILE_BOARD_TILE_TAG);
    document.body.appendChild(dashboardElement);

    const selectedDashboardElement = document.querySelector(TILE_BOARD_TILE_TAG);
    (expect(selectedDashboardElement) as any).toBeInTheDocument();
  });

  it('atlas-tile-board-tile identity', async () => {
    await customElements.whenDefined(TILE_BOARD_TILE_TAG);
    const dashboardElement = document.createElement(TILE_BOARD_TILE_TAG);
    document.body.appendChild(dashboardElement);

    (expect(dashboardElement) as any).toContainHTML('<atlas-tile-board-tile></atlas-tile-board-tile>');
    (expect(dashboardElement) as any).toBeEmptyDOMElement();
  });

  it('supports child elements', async () => {
    await customElements.whenDefined(TILE_BOARD_TILE_TAG);
    const parentElement = document.createElement(TILE_BOARD_TILE_TAG);
    document.body.appendChild(parentElement);

    const romanNumerals = ['I', 'II', 'III', 'IV'];
    romanNumerals
      .map((numeral) => ({ numeral, element: document.createElement('div') }))
      .map(({ numeral, element }) => {
        const elementCopy = element;
        elementCopy.textContent = numeral;
        return elementCopy;
      })
      .forEach((element) => parentElement.appendChild(element));

    (expect(parentElement.childElementCount) as any).toEqual(romanNumerals.length);

    (expect(parentElement) as any).toContainHTML(
      '<atlas-tile-board-tile><div>I</div><div>II</div><div>III</div><div>IV</div></atlas-tile-board-tile>'
    );
    (expect(parentElement.innerHTML) as any).toMatchSnapshot('<div>I</div><div>II</div><div>III</div><div>IV</div>');
  });

});
