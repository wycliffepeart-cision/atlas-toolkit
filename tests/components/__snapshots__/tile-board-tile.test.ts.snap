// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile Board supports child elements: <div>I</div><div>II</div><div>III</div><div>IV</div> 1`] = `
<div>
  I
</div>
<div>
  II
</div>
<div>
  III
</div>
<div>
  IV
</div>
`;
